tests/openai/test_openai_agent_autolog.py

```diff
@@ -0,0 +1,245 @@
+    original = oai_client.chat.completions.create
+    expected_responses = copy.deepcopy(expected_responses)
+    with mock.patch.object(oai_client.chat.completions, "create") as mock_create:
+
+        def _mocked_create(*args, **kwargs):
+            # We need to call the original SDK function because OpenAI autolog patches
+            # it to generate a span for completion.
+            original(*args, **kwargs)
+            return expected_responses.pop(0)
+
+        mock_create.side_effect = _mocked_create
+
+        yield
+
+
+def test_autolog_swarm_agent(client):
+    mlflow.openai.autolog()
+
+    # NB: We have to mock the OpenAI SDK responses to make agent works
+    _DUMMY_RESPONSES = [
+        _get_chat_completion(tool_call="transfer_to_spanish_agent"),
+        _get_chat_completion(content="¡Hola! Estoy bien, gracias. ¿Y tú, cómo estás?"),
+    ]
+
+    swarm = Swarm(client=client)
+    english_agent = Agent(name="English_Agent", instructions="You only speak English")
+    spanish_agent = Agent(name="Spanish_Agent", instructions="You only speak Spanish")
+
+    def transfer_to_spanish_agent():
+        """Transfer spanish speaking users immediately"""
+        return spanish_agent
+
+    english_agent.functions.append(transfer_to_spanish_agent)
+
+    with mock_openai(client, expected_responses=_DUMMY_RESPONSES):
+        messages = [{"role": "user", "content": "Hola.  ¿Como estás?"}]
+        response = swarm.run(agent=english_agent, messages=messages)
+
+    assert response.messages[-1]["content"] == "¡Hola! Estoy bien, gracias. ¿Y tú, cómo estás?"
+    traces = get_traces()
+    assert len(traces) == 1
+    trace = traces[0]
+    assert trace.info.status == "OK"
+    spans = trace.data.spans
+    assert len(spans) == 6  # 1 root + 1 function call + 2 get_chat_completion + 2 Completions
+    assert spans[0].name == "run"
+    assert spans[0].inputs["agent"]["name"] == "English_Agent"
+    assert spans[0].inputs["messages"] == messages
+    assert (
+        spans[0].outputs["messages"][-1]["content"]
+        == "¡Hola! Estoy bien, gracias. ¿Y tú, cómo estás?"
+    )
+    assert spans[1].name == "English_Agent.get_chat_completion"
+    assert spans[1].parent_id == spans[0].span_id
+    assert spans[2].name == "Completions_1"
+    assert spans[2].parent_id == spans[1].span_id
+    assert spans[3].name == "English_Agent.transfer_to_spanish_agent"
+    assert spans[3].span_type == SpanType.TOOL
+    assert spans[3].inputs == {}
+    assert spans[3].outputs["name"] == "Spanish_Agent"
+    assert spans[3].parent_id == spans[0].span_id
+    assert spans[4].name == "Spanish_Agent.get_chat_completion"
+    assert spans[4].parent_id == spans[0].span_id
+    assert spans[5].name == "Completions_2"
+    assert spans[5].parent_id == spans[4].span_id
+
+
+def test_autolog_swarm_agent_with_context_variables(client):
+    from swarm import Agent, Swarm
+
+    mlflow.openai.autolog()
+
+    _DUMMY_RESPONSES = [
```

Comment:
```suggestion
    dummy_responses = [
```

`_DUMMY_RESPONSES` looks like "This variable is not used.".